ddtrace/internal/writer.py

```diff
@@ -355,21 +359,27 @@ def recreate(self):
         )
 
     def _put(self, data, headers):
-        conn = get_connection(self.agent_url, self._timeout)
-
+        # type: (bytes, Dict[str, str]) -> Response
         with StopWatch() as sw:
+            if self._conn is None:
+                self._conn = get_connection(self.agent_url, self._timeout)
```

Comment:
Look into timeout to see if it affects keep-alive

Reply:
From the [Python docs](https://docs.python.org/3/library/http.client.html#http.client.HTTPConnection):

> If the optional timeout parameter is given, blocking operations (like connection attempts) will timeout after that many seconds (if it is not given, the global default timeout setting is used). 

So doesn't look like it affects keep alive and it should be the server that dictates how long a connection is to be kept alive (https://developer.mozilla.org/en-US/docs/Web/HTTP/Connection_management_in_HTTP_1.x)